apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2445,6 +2445,91 @@ static void tc_libc_stdio_stdsostream(void)
 }
 
 /**
+* @testcase         tc_libc_stdio_tempnam_tc
+* @brief            Returns a pointer to a string that is a valid filename
+* @scenario         The tempnam() function returns a pointer to a unique temporary filename, or NULL if a unique name cannot be generated.
+* @apicovered       tempnam(), mkstemp(), mktemp()
+* @precondition     NA
+* @postcondition    NA
+*/
+
+static void tc_libc_stdio_tempnam_tc(void)
+{
+	const char filename[] = "tempnam_tc_test";
+	char *ret;
+	int ret_check;
+	FILE *fp;
+
+	ret_check = mount(MOUNT_DEV_DIR, CONFIG_LIBC_TMPDIR, "smartfs", 0, NULL);
+	TC_ASSERT_EQ("mount", ret_check, OK);
+
+	ret = tempnam(CONFIG_LIBC_TMPDIR, filename);
+	TC_ASSERT_NEQ("tempnam", ret, NULL);
+
+	fp = fopen(ret, "w");
+	TC_ASSERT_NEQ("fopen", fp, NULL);
+	fclose(fp);
+
+	ret = tempnam(CONFIG_LIBC_TMPDIR, NULL);
+	TC_ASSERT_NEQ("tempnam", ret, NULL);
+
+	fp = fopen(ret, "w");
+	TC_ASSERT_NEQ("fopen", fp, NULL);
+	fclose(fp);
+
+#ifdef CONFIG_FS_WRITABLE
+	ret_check = mktemp(ret);
+	TC_ASSERT_EQ_CLEANUP("mktemp", ret_check, ERROR, goto errout);
+	ret_check = mktemp("/tmp/-XXX.tmp");
+	TC_ASSERT_EQ_CLEANUP("mktemp", ret_check, OK, goto errout);
+#endif
+
+	umount(CONFIG_LIBC_TMPDIR);
+
+	TC_SUCCESS_RESULT();
+
+errout:
```

Comment:
It is used when CONFIG_FS_WRITABLE is enabled.
And is it needed? It  seems to be unnessasary.
I mean this can be put directly

Reply:
As a result of double check, umount should be added for all macro in this function excluding mount fail, line 2464 and 2511

Reply:
ok will do that